pyomo/contrib/fbbt/interval.py

```diff
@@ -92,13 +101,111 @@ def power(xl, xu, yl, yu):
                 else:
                     lb = xl ** y
                     ub = xu ** y
+    elif yl == yu:
+        # the exponent is allowed to be fractional, so x must be positive
+        xl = 0
+        lb, ub = power(xl, xu, yl, yu)
     else:
-        lb = -math.inf
-        ub = math.inf
+        msg = 'encountered an exponent where the base is allowed to be negative '
+        msg += 'and the exponent is allowed to be fractional and is not fixed. '
+        msg += 'Assuming the lower bound of the base to be 0.'
+        warnings.warn(msg)
+        logger.warning(msg)
+        xl = 0
+        lb, ub = power(xl, xu, yl, yu)
 
     return lb, ub
 
 
+def _inverse_power1(zl, zu, yl, yu, orig_xl, orig_xu):
+    """
+    z = x**y => compute bounds on x
+    x = exp(ln(z) / y)
+    """
+    xl, xu = log(zl, zu)
+    xl, xu = div(xl, xu, yl, yu)
+    xl, xu = exp(xl, xu)
+
+    # if y is an integer, then x can be negative
+    if yl == yu and yl == round(yl):
+        y = yl
+        if y == 0:
+            xl = -math.inf
+            xu = math.inf
+        elif y % 2 == 0:
+            if zu < 0:
+                raise ValueError('Infeasible. Anything to the power of {0} must be positive.'.format(y))
+            if y > 0:
+                if zu == 0:
+                    _xl = 0
+                    _xu = 0
+                elif zl <= 0:
+                    _xl = -xu
+                    _xu = xu
+                else:
+                    if orig_xl <= -xl:
+                        _xl = -xu
+                    else:
+                        _xl = xl
+                    if orig_xu < xl:
+                        _xu = -xl
+                    else:
+                        _xu = xu
+                xl = _xl
+                xu = _xu
+            else:
+                if zu == 0:
+                    raise ValueError('Infeasible.')
+                elif zl <= 0:
+                    _xl = -math.inf
+                    _xu = math.inf
+                else:
+                    if orig_xl <= -xl:
+                        _xl = -xu
+                    else:
+                        _xl = xl
+                    if orig_xu < xl:
+                        _xu = -xl
+                    else:
+                        _xu = xu
+                xl = _xl
+                xu = _xu
+        elif y % 2 == 1:
```

Comment:
This should probably be:
```
else: # y % 2 == 1
```
(so folks digging through the code understand that there aren't unhandled cases in this if tree)

Reply:
Good suggestion. I will fix this.